### DISCUSSION TITLE
Ignoring warning emitted by libs in specific folder (and its subfolders) using `pytest`

### DISCUSSION THREAD

**Discussion by HardMax71** [pytest-dev/pytest#12811] [Q&A]
Hi all, 

I'm using `pytest` lib for testing and sometimes I get warnings emitted in code in libs imported (all are in `/venv` directory).

I do not want to turn off warnings that are emitted in code I wrote, but warnings emitted in code of any 3rd party lib are of no use.

How to turn off or hide all warnings that come from code in specified directory, as well in any subfolders of that directory, in my case - `/venv`?

One specific example for my use case can be:
```
============================================ warnings summary ============================================ 
test_auth.py: 3 warnings
  C:\Users\User\<my project>\security.py:33: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    expire = datetime.utcnow() + timedelta(days=settings.REFRESH_TOKEN_EXPIRE_DAYS)
===
^ this line is in code I wrote, warnings should be turned on.
===

test_auth.py: 2 warnings
  C:\Users\User\<my project>\venv\Lib\site-packages\jose\jwt.py:311: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    now = timegm(datetime.utcnow().utctimetuple())
===
^ this line is somewhere in python-jwt lib in /venv, I dont want to see them at all. 
===
```

Also asked this question on [SO](https://stackoverflow.com/questions/78974606/ignoring-warning-emitted-by-libs-in-venv-using-pytest?noredirect=1#comment139246750_78974606), so far no answers :(

**Comment by kwist-sgr** [comment]
I`m not sure you can ignore warnings by path, but you can ignore them for a specific package
```TOML
# pyproject.toml (also you can use pytest.ini)

[tool.pytest.ini_options]
filterwarnings = [
  "ignore:::jose.*",
  "ignore:::hose.*",
]
```

**Reply by HardMax71** [reply]
Yay, I do know about that option, but it will also turn off warnings from `jose.*` libs emitted in user code, I don't really want that.

**Reply by kwist-sgr** [reply]
I guess, technically, this can be done, but only by the monkey patching :man_shrugging:
